plugins/outputs/postgresql/postgresql_test.go

```diff
@@ -0,0 +1,623 @@
+		la.cond.Wait()
+	}
+	return la.logs[:]
+}
+
+// Waits for a specific query log from pgx to show up.
+func (la *LogAccumulator) WaitFor(f func(l Log) bool, waitCommit bool) {
+	la.cond.L.Lock()
+	defer la.cond.L.Unlock()
+	i := 0
+	var commitPid uint32
+	for {
+		for ; i < len(la.logs); i++ {
+			log := la.logs[i]
+			if commitPid == 0 {
+				if f(log) {
+					if !waitCommit {
+						return
+					}
+					commitPid = log.args[1].(MSI)["pid"].(uint32)
+				}
+			} else {
+				if len(log.args) < 2 {
+					continue
+				}
+				data, ok := log.args[1].(MSI)
+				if !ok || data["pid"] != commitPid {
+					continue
+				}
+				if log.args[0] == "Exec" && data["sql"] == "commit" {
+					return
+				} else if log.args[0] == "Exec" && data["sql"] == "rollback" {
+					// transaction aborted, start looking for another match
+					commitPid = 0
+				} else if log.level == pgx.LogLevelError {
+					commitPid = 0
+				}
+			}
+		}
+		la.cond.Wait()
+	}
+}
+
+func (la *LogAccumulator) WaitForQuery(str string, waitCommit bool) {
+	la.WaitFor(func(log Log) bool {
+		return log.format == "PG %s - %+v" &&
+			(log.args[0].(string) == "Query" || log.args[0].(string) == "Exec") &&
+			strings.Contains(log.args[1].(MSI)["sql"].(string), str)
+	}, waitCommit)
+}
+
+func (la *LogAccumulator) WaitForCopy(tableName string, waitCommit bool) {
+	la.WaitFor(func(log Log) bool {
+		return log.format == "PG %s - %+v" &&
+			log.args[0].(string) == "CopyFrom" &&
+			log.args[1].(MSI)["tableName"].(pgx.Identifier)[1] == tableName
+	}, waitCommit)
+}
+
+// Clear any stored logs.
+// Do not run this while any WaitFor* operations are in progress.
+func (la *LogAccumulator) Clear() {
+	la.cond.L.Lock()
+	if len(la.logs) > 0 {
+		la.logs = nil
+	}
+	la.cond.L.Unlock()
+}
+
+func (la *LogAccumulator) Logs() []Log {
+	la.cond.L.Lock()
+	defer la.cond.L.Unlock()
+	return la.logs[:]
+}
+
+func (la *LogAccumulator) Errorf(format string, args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelError, format, args)
+}
+
+func (la *LogAccumulator) Error(args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelError, "%v", args)
+}
+
+func (la *LogAccumulator) Debugf(format string, args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelDebug, format, args)
+}
+
+func (la *LogAccumulator) Debug(args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelDebug, "%v", args)
+}
+
+func (la *LogAccumulator) Warnf(format string, args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelWarn, format, args)
+}
+
+func (la *LogAccumulator) Warn(args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelWarn, "%v", args)
+}
+
+func (la *LogAccumulator) Infof(format string, args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelInfo, format, args)
+}
+
+func (la *LogAccumulator) Info(args ...interface{}) {
+	la.tb.Helper()
+	la.append(pgx.LogLevelInfo, "%v", args)
+}
+
+var ctx = context.Background()
+
+func TestMain(m *testing.M) {
```

Comment:
I think you should only do this if we are _NOT_ in `short` testing mode. Otherwise you will rely on the test-machine hosting a PostgreSQL instance and (in worst-case) write data to that existing instance.